modules/service/profile-service.js

```diff
@@ -127,16 +122,16 @@ class ProfileService {
         if (!identityExists) {
             const event = await this.blockchain.subscribeToEvent('IdentityCreated', null, 5 * 60 * 1000, null, eventData => Utilities.compareHexStrings(eventData.profile, this.config.node_wallet));
             if (event) {
-                this._saveIdentity(event.newIdentity);
-                this.logger.notify(`Identity created for node ${this.config.identity}. Identity is ${this.config.erc725Identity}.`);
+                this.blockchain.saveIdentity(event.newIdentity, blockchainId);
+                this.logger.notify(`Identity created for node ${this.config.identity}. Identity is ${event.newIdentity}. For blockchain id: ${blockchainId}.`);
             } else {
                 throw new Error('Identity could not be confirmed in timely manner. Please, try again later.');
             }
         }
 
-        const event = await this.blockchain.subscribeToEvent('ProfileCreated', null, 5 * 60 * 1000, null, eventData => Utilities.compareHexStrings(eventData.profile, this.config.erc725Identity));
+        const event = await this.blockchain.subscribeToEvent('ProfileCreated', null, 5 * 60 * 1000, null, eventData => Utilities.compareHexStrings(eventData.profile, this.getIdentity(blockchainId)));
```

Comment:
If we're waiting for an event from this specific blockchain_id, we should add that parameter to the subscribeToEvent function